src/sdl2/messagebox.rs

```diff
@@ -48,3 +101,60 @@ pub fn show_simple_message_box(flags: MessageBoxFlag, title: &str,
         Err(SdlError(get_error()))
     }
 }
+
+/// returns the id of the button triggered
+pub fn show_message_box<'a>(flags:MessageBoxFlag, buttons:&'a [ButtonData], title:&str,
+    message:&str, window:Option<&WindowRef>, scheme:Option<MessageBoxColorScheme>)
+    -> Result<i32,ShowMessageError> {
+    use self::ShowMessageError::*;
+    let mut button_id : c_int = 0;
+    let result = unsafe {
```

Comment:
The unsafe block should generally be as small as possible; this is covering large areas of code that don't need to be unsafe.